src/app/course/services/CoursesService.ts

```diff
@@ -0,0 +1,39 @@
+// @ts-ignore
+import CoursesList from './cources.json';
+import {ICourse, TCourse} from "../models/course";
+
+export class CoursesService {
+
+	public async getList(params?: { search: string }): Promise<ICourse[]> {
+		const arr: [] = CoursesList;
+		if (params && params.search) {
+			const {search} = params;
+			return arr.filter((course: ICourse) => {
+				const regexp = new RegExp(search.toLowerCase());
```

Comment:
Using regExp here is overhead and you are repeating the same action in line 12 for every loop. It might be optimized.

Reply:
what about U?

Reply:
What U?